python/ray/data/read_api.py

```diff
@@ -252,6 +252,28 @@ def read_datasource(
         Dataset holding the data read from the datasource.
     """
     ctx = DatasetContext.get_current()
+
+    if ray_remote_args is None:
+        ray_remote_args = {}
+
+    paths = read_args.get("paths", None)
+    if paths and _is_local_scheme(paths):
+        if ray.util.client.ray.is_connected():
+            raise ValueError(
+                f"The local scheme paths {paths} are not supported in Ray Client."
+            )
+        ray_remote_args["scheduling_strategy"] = NodeAffinitySchedulingStrategy(
+            ray.get_runtime_context().get_node_id(),
+            soft=False,
+        )
+        read_args["local_uri"] = True
```

Comment:
I don't think we should pass this via reader args, how about as a separate arg to the data source?

Reply:
This local URI scheme is relevant for data from filesystem, but datasource needs to support beyond that, particularly databases. To avoid messing up `read_args`, I added a separate parameter to the reader constructor.